examples/all-clusters-app/linux/main-common.cpp

```diff
@@ -50,297 +50,43 @@
-    }
-    else if (eventId == Clusters::Switch::Events::MultiPressComplete::Id)
-    {
-        Clusters::SwitchServer::Instance().OnMultiPressComplete(endpoint, newPosition, count);
-    }
-    else
-    {
-        ChipLogError(DeviceLayer, "Unknow event ID:%d", eventId);
-    }
-}
-
-// when the shell is enabled, don't intercept signals since it prevents the user from
-// using expected commands like CTRL-C to quit the application. (see issue #17845)
-// We should stop using signals for those faults, and move to a different notification
-// means, like a pipe. (see issue #19114)
-#if !defined(ENABLE_CHIP_SHELL)
-void OnRebootSignalHandler(int signum)
+void * EventCommandListenerTask(void * arg)
 {
-    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
+    int fd;
+    char readbuf[CHIP_EVENT_COMMAND_SIZE];
+    ssize_t readBytes;
 
-    // The BootReason attribute SHALL indicate the reason for the Node’s most recent boot, the real usecase
-    // for this attribute is embedded system. In Linux simulation, we use different signals to tell the current
-    // running process to terminate with different reasons.
-    BootReasonType bootReason = BootReasonType::kUnspecified;
-    switch (signum)
+    for (;;)
     {
-    case SIGVTALRM:
-        bootReason = BootReasonType::kPowerOnReboot;
-        break;
-    case SIGALRM:
-        bootReason = BootReasonType::kBrownOutReset;
-        break;
-    case SIGILL:
-        bootReason = BootReasonType::kSoftwareWatchdogReset;
-        break;
-    case SIGTRAP:
-        bootReason = BootReasonType::kHardwareWatchdogReset;
-        break;
-    case SIGIO:
-        bootReason = BootReasonType::kSoftwareUpdateCompleted;
-        break;
-    case SIGINT:
-        bootReason = BootReasonType::kSoftwareReset;
-        break;
-    default:
-        IgnoreUnusedVariable(bootReason);
-        ChipLogError(NotSpecified, "Unhandled signal: Should never happens");
-        chipDie();
-        break;
-    }
-
-    Server::GetInstance().DispatchShutDownAndStopEventLoop();
-}
-
-void OnSoftwareFaultSignalHandler(int signum)
-{
-    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
-
-    VerifyOrDie(signum == SIGUSR1);
-    PlatformMgr().ScheduleWork(HandleSoftwareFaultEvent);
-}
-
-void OnGeneralFaultSignalHandler(int signum)
-{
-    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
-
-    uint32_t eventId;
-    switch (signum)
-    {
-    case SIGUSR2:
-        eventId = Clusters::GeneralDiagnostics::Events::HardwareFaultChange::Id;
-        break;
-    case SIGHUP:
-        eventId = Clusters::GeneralDiagnostics::Events::RadioFaultChange::Id;
-        break;
-    case SIGTTIN:
-        eventId = Clusters::GeneralDiagnostics::Events::NetworkFaultChange::Id;
-        break;
-    default:
-        ChipLogError(NotSpecified, "Unhandled signal: Should never happens");
-        chipDie();
-        break;
-    }
+        fd = open(ChipEventFifoPath, O_RDONLY);
```

Comment:
need error checking for file open errors. Maybe exit if open fails (this would allow the main thread to kill this thread).

Reply:
Acked